src/Flux/utils.jl

```diff
@@ -0,0 +1,18 @@
+function destructure(m)
+  xs = []
+  mapleaves(m) do x
+    x isa TrackedArray && push!(xs, x)
+    return x
+  end
+  return vcat(vec.(xs)...)
```

Comment:
use `mapreduce(vec, vcat, xs)`
It is faster, due to the `reduce(vcat, xs)` hitting an optimized method

Reply:
It'd be best to time this in this case, since we're going to hit Flux's gradient definition anyway, and that potentially involves a lot of extra work.

In any case I'm not too worried about micro-optimising this, it's not going to be as fast as a real solution but just needs to work so we can prototype things.